### ISSUE TITLE
What's the value of FLAGS.label_smooth?

### ISSUE THREAD

**Issue by lukk47** [P2333/Adaptive-Diversity-Promoting#5]
I didn't find the default value for the label_smooth which are used in function '_Loss_withEE_DPP', and why do you need this smoother for the cross entropy loss function? Thanks!

**Comment by P2333** [comment]
The label_smooth value is controlled by `FLAGS.lamda`, where we set `FLAGS.lamda = 2` in our paper. Other values can also be set depending on the degree of label smoothing.

The main reason for introducing the smoother is to reserve numerical significance for non-maximal predictions, such that we can further regularize the non-maximal predictions with the ensemble diversity (ED) term. Namely, if we do not use the smoother, then the optimal prediction will tend to one-hot and ignore the regularization induced by ED, as claimed in Theorem 1 in [our paper](http://proceedings.mlr.press/v97/pang19a/pang19a.pdf).

**Comment by lukk47** [comment]
Thanks for replying! And I am sorry I didn't explain myself clear. I think you are mentioning the [EE (Ensemble Entropy) loss](https://github.com/P2333/Adaptive-Diversity-Promoting/blob/d5648e958465d1db6cff4e08c9f2f51fdd61a95a/advtrain_cifar10.py#L150) under control of lambda. I am curious about [FLAGS.label_smooth](https://github.com/P2333/Adaptive-Diversity-Promoting/blob/d5648e958465d1db6cff4e08c9f2f51fdd61a95a/advtrain_cifar10.py#L140) which is used to generate a soft ground truth for computing the [classification loss](https://github.com/P2333/Adaptive-Diversity-Promoting/blob/d5648e958465d1db6cff4e08c9f2f51fdd61a95a/advtrain_cifar10.py#L149).
What's the reason behind to use the soft ground truth for the classification loss and what's the default value of FLAG.label_smooth? Thank you!

**Comment by P2333** [comment]
Oh, this is embarrassing... I did not remember that there is a label_smooth flag. I think I used this flag to perform ablation studies, and you can directly set it to 1 (no label smooth). 

B.T.W, I will release an updated version of code which are better polished and written in PyTorch, you may resort to the new code then.

**Comment by lukk47** [comment]
Wow, great. Looking forward to the PyTorch code.
